### PR TITLE
Clarify that you can set channel welcome message & fix JSON indentation

### PR DESCRIPTION
**Pull request by bbodenmiller**
#### Summary

Clarify that you can set channel welcome message

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 **Maintainer:** [@jfrerich](https://github.com/jfrerich)
 **Co-Maintainer:** [@iomodo](https://github.com/iomodo)
 
-Use this plugin to improve onboarding and HR processes. It adds a Welcome Bot that helps add new team members to channels.
+Use this plugin to improve onboarding and HR processes. It adds a Welcome Bot that helps welcome users to team and/or channels as well as easily join channels based on selections.
```
Comment (justinegeffen) [inline README.md:11]:
```suggestion
Use this plugin to improve onboarding and HR processes. It adds a Welcome Bot that helps welcome users to teams and/or channels as well as easily join channels based on selections.
```

File: README.md [hunk 2]
```diff
@@ -74,11 +74,11 @@ where
     - **ActionSuccessfulMessage**: Message posted after the user takes this action and joins the specified channels.
     - **ChannelsAddedTo**: List of channel names the user is added to. Must be the channel handle used in the URL, in lowercase. For example, in the following URL the **channel name** value is `my-channel`: https://example.com/my-team/channels/my-channel
 
-The preview of the configured messages can be done via bot commands:
+The preview of the configured messages as well as creating channel welcome message can be done via bot commands:
```
Comment (justinegeffen) [inline README.md:77]:
```suggestion
The preview of the configured messages, as well as the creation of a channel welcome message, can be done via bot commands:
```

File: README.md [hunk 3]
```diff
@@ -96,93 +96,93 @@ Moreover, those who join the DevSecOps team should automatically be added to Esc
 To accomplish the above, you can specify the following configuration in your config.json file.
```
Comment (justinegeffen) [inline README.md:96]:
```suggestion
To accomplish the above, you can specify the following configuration in your `config.json` file.
```

File: README.md [hunk 4]
```diff
@@ -96,93 +96,93 @@ Moreover, those who join the DevSecOps team should automatically be added to Esc
...
-                        "ChannelsAddedTo": ["leads", "sales-discussion", "win-loss-analysis"],
-                        "ActionSuccessfulMessage": [
-                            "### Awesome! I have added you to the following developer channels:",
-                            "~leads - To stay updated on incoming leads",
-                            "~sales-discussion - To collaborate with your fellow account managers",
-                            "~win-loss-analysis - To conduct win-loss analysis of closed deals"
+                        "TeamName": "staff",
+                        "DelayInSeconds": 10,
+                        "AttachmentMessage": [
+                            "Let's get started by adding you to key channels! What is your role in the company?"
```
Comment (justinegeffen) [inline README.md:117]:
```suggestion
                            "Let's get started by adding you to key channels! What's your role in the company?"
```

File: README.md [hunk 5]
```diff
@@ -96,93 +96,93 @@ Moreover, those who join the DevSecOps team should automatically be added to Esc
...
+                        ],
+                        "Actions" : [
+                            {
+                                "ActionType": "button",
+                                "ActionDisplayName": "Developer",
+                                "ActionName": "developer-action",
+                                "ChannelsAddedTo": ["bugs", "jira-tasks", "sprint-planning"],
+                                "ActionSuccessfulMessage": [
+                                    "### Awesome! I have added you to the following developer channels:",
```
Comment (justinegeffen) [inline README.md:126]:
```suggestion
                                    "### Awesome! I've added you to the following developer channels:",
```

File: README.md [hunk 6]
```diff
@@ -96,93 +96,93 @@ Moreover, those who join the DevSecOps team should automatically be added to Esc
...
+                                    "~sprint-planning - To plan and manage your team's Jira sprint"
+                                ]
+                            },
+                            {
+                                "ActionType": "button",
+                                "ActionDisplayName": "Account Manager",
+                                "ActionName": "account-manager-action",
+                                "ChannelsAddedTo": ["leads", "sales-discussion", "win-loss-analysis"],
+                                "ActionSuccessfulMessage": [
+                                    "### Awesome! I have added you to the following developer channels:",
```
Comment (justinegeffen) [inline README.md:138]:
```suggestion
                                    "### Awesome! I've added you to the following developer channels:",
```

File: README.md [hunk 7]
```diff
@@ -96,93 +96,93 @@ Moreover, those who join the DevSecOps team should automatically be added to Esc
...
+                                    "~win-loss-analysis - To conduct win-loss analysis of closed deals"
+                                ]
+                            },
+                            {
+                                "ActionType": "button",
+                                "ActionDisplayName": "Support",
+                                "ActionName": "support-action",
+                                "ChannelsAddedTo": ["bugs", "customer-support", "leads"],
+                                "ActionSuccessfulMessage": [
+                                    "### Awesome! I have added you to the following developer channels:",
```
Comment (justinegeffen) [inline README.md:150]:
```suggestion
                                    "### Awesome! I've added you to the following developer channels:",
```

File: README.md [hunk 8]
```diff
@@ -96,93 +96,93 @@ Moreover, those who join the DevSecOps team should automatically be added to Esc
...
+                                    "~bugs - To help investigate or report bugs",
+                                    "~customer-support - To troubleshoot and resolve customer issues",
+                                    "~leads - To discuss potential accounts with other account managers"
```
Comment (justinegeffen) [inline README.md:153]:
```suggestion
                                    "~leads - To discuss potential accounts with other Customer Engineers"
```
@jasonblais, is this the right role reference?

File: README.md [hunk 9]
```diff
@@ -96,93 +96,93 @@ Moreover, those who join the DevSecOps team should automatically be added to Esc
...
-                            "### Awesome! I have added you to the following developer channels:",
-                            "~bugs - To help investigate or report bugs",
-                            "~customer-support - To troubleshoot and resolve customer issues",
-                            "~leads - To discuss potential accounts with other account managers"
+                        "TeamName": "devsecops",
+                        "DelayInSeconds": 5,
+                        "Message": [
+                            "### Welcome {{.UserDisplayName}} to the {{.Team.DisplayName}} team!",
+                            "",
+                            "**If you are not a member of the Security Meta Team and ended up on this team by accident, please report this issue and leave the team!**",
```
Comment (justinegeffen) [inline README.md:164]:
```suggestion
                            "**If you're not a member of the Security Meta Team and ended up on this team by accident, please report this issue and leave the team!**",
```

File: README.md [hunk 10]
```diff
@@ -96,93 +96,93 @@ Moreover, those who join the DevSecOps team should automatically be added to Esc
...
+                            "",
+                            "##### I've added you to a few channels to get you started:",
+                            "",
+                            "~escalation-process - To review the DevSecOps escalation process",
+                            "~incidents - To collaborate and resolve seucrity incidents"
```
Comment (justinegeffen) [inline README.md:169]:
```suggestion
                            "~incidents - To collaborate on and resolve security incidents"
```

**Review by justinegeffen — COMMENTED** [review]
Added some suggestions and need to confirm whether "account manager" is the correct role.

**Review by jfrerich — APPROVED** [review]
Thank you @bbodenmiller!  LGTM, but please merge @justinegeffen's suggestions 🚀

**Comment by bbodenmiller** [comment]
@jfrerich applied all but one because it's just an example and I don't think `Customer Engineer` is a widely used role name.

**Comment by jasonblais** [comment]
@justinegeffen ready for re-review

**Comment by jfrerich** [comment]
@justinegeffen, gentle reminder to review this PR :)

File: README.md [hunk 11]
```diff
@@ -74,11 +74,11 @@ where
     - **ActionSuccessfulMessage**: Message posted after the user takes this action and joins the specified channels.
     - **ChannelsAddedTo**: List of channel names the user is added to. Must be the channel handle used in the URL, in lowercase. For example, in the following URL the **channel name** value is `my-channel`: https://example.com/my-team/channels/my-channel
 
-The preview of the configured messages can be done via bot commands:
+The preview of the configured messages, as well as the creation of a channel welcome message, can be done via bot commands:
 * `/welcomebot help` - show a short usage information
```
Comment (justinegeffen) [inline README.md:78]:
```suggestion
* `/welcomebot help` - Displays usage information
```

File: README.md [hunk 12]
```diff
@@ -74,11 +74,11 @@ where
...
 * `/welcomebot list` - lists the teams for which greetings were defined
```
Comment (justinegeffen) [inline README.md:79]:
```suggestion
* `/welcomebot list` - Lists the teams for which greetings were defined
```

File: README.md [hunk 13]
```diff
@@ -74,11 +74,11 @@ where
...
 * `/welcomebot preview [team-name]` - sends ephemeral messages to the user calling the command, with the preview of the welcome message[s] for the given team name and the user that requested the preview
```
Comment (justinegeffen) [inline README.md:80]:
```suggestion
* `/welcomebot preview [team-name]` - Sends ephemeral messages to the user calling the command, with the preview of the welcome message[s] for the given team name and the user that requested the preview.
```

File: README.md [hunk 14]
```diff
@@ -74,11 +74,11 @@ where
...
-* `/welcomebot set_channel_welcome` - sets the given text as current's channel welcome message
+* `/welcomebot set_channel_welcome [welcome-message]` - sets the given text as current's channel welcome message
```
Comment (justinegeffen) [inline README.md:81]:
```suggestion
* `/welcomebot set_channel_welcome [welcome-message]` - Sets the given text as current's channel welcome message.
```

File: README.md [hunk 15]
```diff
@@ -74,11 +74,11 @@ where
...
 * `/welcomebot get_channel_welcome` - gets the current's channel welcome message
```
Comment (justinegeffen) [inline README.md:82]:
```suggestion
* `/welcomebot get_channel_welcome` - Gets the current channel's welcome message.
```

File: README.md [hunk 16]
```diff
@@ -74,11 +74,11 @@ where
...
 * `/welcomebot delete_channel_welcome` - deletes the current's channel welcome message
```
Comment (justinegeffen) [inline README.md:83]:
```suggestion
* `/welcomebot delete_channel_welcome` - Deletes the current channel'S welcome message.
```

File: README.md [hunk 17]
```diff
@@ -93,96 +93,96 @@ Those who join the Staff team should be added to a set of channels based on thei
...
+                            },
+                            {
+                                "ActionType": "button",
+                                "ActionDisplayName": "Account Manager",
+                                "ActionName": "account-manager-action",
+                                "ChannelsAddedTo": ["leads", "sales-discussion", "win-loss-analysis"],
+                                "ActionSuccessfulMessage": [
+                                    "### Awesome! I've added you to the following developer channels:",
+                                    "~leads - To stay updated on incoming leads",
+                                    "~sales-discussion - To collaborate with your fellow account managers",
```
Comment (justinegeffen) [inline README.md:140]:
```suggestion
                                    "~sales-discussion - To collaborate with your fellow Customer Engineers,
```

File: README.md [hunk 18]
```diff
@@ -93,96 +93,96 @@ Those who join the Staff team should be added to a set of channels based on thei
...
+                                "ActionSuccessfulMessage": [
+                                    "### Awesome! I've added you to the following developer channels:",
+                                    "~bugs - To help investigate or report bugs",
+                                    "~jira-tasks - To stay updated on Jira tasks",
+                                    "~sprint-planning - To plan and manage your team's Jira sprint"
+                                ]
+                            },
+                            {
+                                "ActionType": "button",
+                                "ActionDisplayName": "Account Manager",
```
Comment (justinegeffen) [inline README.md:134]:
```suggestion
                                "ActionDisplayName": "Customer Engineer",
```

File: README.md [hunk 19]
```diff
@@ -93,96 +93,96 @@ Those who join the Staff team should be added to a set of channels based on thei
...
+                                "ActionName": "account-manager-action",
```
Comment (justinegeffen) [inline README.md:135]:
```suggestion
                                "ActionName": "customer-engineer-action",
```

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**